api/nodes/views.py

```diff
@@ -907,6 +907,53 @@ def get_parser_context(self, http_request):
         return res
 
 
+class NodeLinkedByNodesList(JSONAPIBaseView, generics.ListAPIView, NodeMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        ContributorOrPublic,
+        ExcludeWithdrawals,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_BASE_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    view_category = 'nodes'
+    view_name = 'node-linked-by-nodes'
+
+    serializer_class = NodeSerializer
+
+    def get_queryset(self):
+        node = self.get_node()
+        auth = get_user_auth(self.request)
+        node_relations = NodeRelation.objects.filter(child=node, is_node_link=True)\
+            .select_related('parent').exclude(parent__type='osf.collection').exclude(parent__type='osf.registration')
+        return [each.parent for each in node_relations if each.parent.can_view(auth)]
```

Comment:
Can we not do this without a list comprehension?

Reply:
Talked to @sloria, and it seemed to be the only way (well, iterating through the results that is, not necessarily a list comprehension). Am I wrong in that assumption Steve?